components/scream/src/physics/share/physics_functions.hpp

```diff
@@ -112,18 +112,30 @@ struct Functions
   // The result is the potential temperature, units in K
   // The inputs are
   //   T_mid is the atmospheric temperature, units in K
-  //   Exners function, unitless.  Exners function can be derived using `get_exner` defined above.
+  //   p_mid is the atmospheric pressure, units in Pa.  Used with Exners function using `get_exner` defined above.
   KOKKOS_FUNCTION
-  static Spack T_to_th(const Spack& T_atm, const Spack& exner, const Smask& range_mask);
+  static Spack get_potential_temperature(const Spack& T_mid, const Spack& p_mid, const Smask& range_mask);
 
+  KOKKOS_FUNCTION
+  template <typename InputProvider>
+  static void get_potential_temperature(const int nlev,
+                                        const InputProvider& T_mid,
+                                        const InputProvider& p_mid,
+                                        const view_1d<Scalar>& T_potential);
   // Converts potential temperature into temperature
   // The result is the temperature, units in K
   // The inputs are
   //   Potential Temperature, units in K
-  //   Exners function, unitless.  Exners function can be derived using `get_exner` defined above.
+  //   p_mid is the atmospheric pressure, units in Pa.  Used with Exners function using `get_exner` defined above.
   KOKKOS_FUNCTION
-  static Spack th_to_T(const Spack& th_atm, const Spack& exner, const Smask& range_mask);
+  static Spack get_potential_temperature_inv(const Spack& th_mid, const Spack& p_mid, const Smask& range_mask);
 
+  KOKKOS_FUNCTION
+  template <typename InputProvider>
+  static void get_potential_temperature_inv(const int nlev,
```

Comment:
Personally, I'm not a fan of the fcn name. If nothing, b/c Theta is a fcn of two variables. How about `compute_temperature` or `Theta_to_T`?

Reply:
Similar response as the `T_to_Theta` discussion above.

Reply:
If we keep `get_potential_temperature`, which is fine, I would call this fcn `get_temperature`. The "_inv" suffix only makes sense for fcns of one variable only.